README.md

```diff
@@ -123,5 +123,146 @@ LabelCell(key: "company", state: LabelState(text: "Shopify",
+profilePicture.widthAnchor.constraint(equalToConstant: 60),
+profilePicture.leadingAnchor.constraint(equalTo: leadingAnchor),
+profilePicture.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+profilePicture.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
+stackView.centerYAnchor.constraint(equalTo: profilePicture.centerYAnchor),
+stackView.topAnchor.constraint(greaterThanOrEqualTo: topAnchor),
+stackView.bottomAnchor.constraint(lessThanOrEqualTo: bottomAnchor),
+stackView.leadingAnchor.constraint(equalTo: profilePicture.trailingAnchor, constant: 12),
+stackView.trailingAnchor.constraint(lessThanOrEqualTo: timeStampLabel.leadingAnchor, constant: -12),
+timeStampLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+timeStampLabel.topAnchor.constraint(equalTo: topAnchor)
+]
+NSLayoutConstraint.activate(constraints)
+
+profilePicture.layer.borderWidth = 1
+profilePicture.layer.borderColor = UIColor.black.cgColor
+profilePicture.layer.cornerRadius = 30
+profilePicture.backgroundColor = UIColor.gray
+
+nameLabel.text = "FirstName LastName"
+nameLabel.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+messageLabel.text = "This is a new message"
+messageLabel.font = UIFont.systemFont(ofSize: 15, weight: .light)
+
+timeStampLabel.font = UIFont.systemFont(ofSize: 12, weight: .light)
+timeStampLabel.text = "Date Label"
+timeStampLabel.setContentHuggingPriority(.required, for: .horizontal)
+}
+}
+```
+
+Next, we'll build the `MessageViewState` struct that will take care of setting all the views properties:
+
+```
+struct Message {
+let name: String
+let image: UIImage
+let date: Date
+let text: String
+}
+
+struct MessageViewState: Equatable {
+
+let date: Date
+let name: String
+let message: String
+let image: UIImage
+
+init(message: Message) {
+self.date = message.date
+self.name = message.name
+self.message = message.text
+self.image = message.image
+}
+
+static func updateView(_ view: MessageView, state: MessageViewState?) {
+guard let state = state else { return }
+view.timeStampLabel.text = "\(state.date)"
+view.nameLabel.text = state.name
+view.messageLabel.text = state.message
+view.profilePicture.image = state.image
+}
+
+}
+```
+
+Finally, the `MessageListBuilder` will accept an array of `Messages` and return the appropriate `TableSection`. We will define a custom `HostCell` called `MessageCell` and pass in our custom `MessageView`, `MessageViewState` and layout
+
+```
+struct MessageListBuilder {
+
+//custom HostCell
+private typealias MessageCell = HostCell<MessageView, MessageViewState, LayoutMarginsTableItemLayout>
+
+static func sections(messages: [Message]) -> TableSection {
+var rows = [CellConfigType]()
+for message in messages {
+// each cell should have a unique key
+rows.append(MessageCell(key: "\(message)-\(UUID().uuidString)-cell", state: MessageViewState(message: message)) { (view, state) in
+
+})
+}
+return TableSection(key: "entire-table", rows: rows)
+}
+}
+```
```

Comment:
I think we can skip the builder, we have an example above right? If not, directly just creating the cell without `MessageListBuilder` would be fine

Reply:
I don't see a list builder in the current read me. I think it can be valuable to explain that if you have multiple types of `CellConfigType` it is easy to pass all of that logic to the list builder. What do you think? I can make that clearly in the text

Reply:
What I meant is that a list builder is not part of FunctionalTableData, is just a way for us to centralize the creating of the cells.

I still don't think we should be coupling Models + ListBuilders into this README. Those two are just tied to how we do things, but may not be how people should do it